From 1071268d06da269de56c895af940e6f15bcf79e7 Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Wed, 2 Dec 2015 07:15:43 -0300
Subject: [PATCH] rpc-transport.c: include sys/select.h for fd_set
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

fd_set and friends, according to POSIX.1-2001, needs sys/select.h, so
include it otherwise the build fails for uClibc:

p11-kit/rpc-transport.c: In function ‘rpc_socket_read’:
p11-kit/rpc-transport.c:350:2: error: unknown type name ‘fd_set’
p11-kit/rpc-transport.c:416:4: warning: implicit declaration of function
‘FD_ZERO’ [-Wimplicit-function-declaration]

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 p11-kit/rpc-transport.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/p11-kit/rpc-transport.c b/p11-kit/rpc-transport.c
index 7176b39..5251e11 100644
--- a/p11-kit/rpc-transport.c
+++ b/p11-kit/rpc-transport.c
@@ -55,6 +55,7 @@
 #include <string.h>
 
 #ifdef OS_UNIX
+#include <sys/select.h>
 #include <sys/socket.h>
 #include <sys/wait.h>
 #include <sys/un.h>
-- 
2.4.10

